lib/ecto/type.ex

```diff
@@ -347,7 +360,6 @@ defmodule Ecto.Type do
   defp do_match?(_left, :any),  do: true
   defp do_match?(:any, _right), do: true
   defp do_match?({outer, left}, {outer, right}), do: match?(left, right)
-  defp do_match?({:array, :any}, {:embed, %{cardinality: :many}}), do: true
```

Comment:
Hmm … are we sure we do not need `{:parameterized, Ecto.Type.EmbedMany, _opts}` match here? Not following the code, but I also wonder what's with `{:map, :any}` and `{:parameterized, Ecto.Type.Embed, _opts}` match?

Reply:
I have added a passthrough to a callback on ParameterizedType